qa-pipelines/tasks/cf-teardown.sh

```diff
@@ -30,8 +30,12 @@ for namespace in "$CF_NAMESPACE" "$UAA_NAMESPACE" ; do
     done
 done
 
-cap_platform=$(kubectl get configmap -n kube-system cap-values -o json | jq -r .data.platform)
-if [[ ${cap_platform} == "azure" ]] || [[ ${cap_platform} == "gke" ]]; then
+if [[ ${cap_platform} != "eks" ]]; then
```

Comment:
I see `cap_platform` now gets set to `eks` in `prepare-kubeconfig.sh` which is sourced, but I think it's unset on non-eks.

Regardless, this isn't correct; we don't want to run `azure_dns_clear` for all deployments, just azure and gke deployments.

Reply:
we are using azure dns for EKS as well

Reply:
right, ignore the point about setting the DNS then, I now see that you moved that into a different block just after this one.

What I'm saying is that `cap_platform` won't be set if the platform isn't eks, so this will fail due to `nounset`

Instead this should be:
```
cap_platform=${cap_platform:-$(kubectl get configmap -n kube-system cap-values -o json | jq -r .data.platform)}
```

Reply:
Note that you don't need the `if` clause at all there